carbon_black_cloud/plugin.spec.yaml

```diff
@@ -0,0 +1,456 @@
+      title: "Sensor Kit Type"
+      type: string
+      description: "Sensor kit type"
+      required: false
+    sensorOutOfDate:
+      title: "Sensor Out of Date"
+      type: boolean
+      description: "Sensor out of date"
+      required: false
+    sensorPendingUpdate:
+      title: "Sensor Pending Update"
+      type: boolean
+      description: "Sensor pending update"
+      required: false
+    sensorStates:
+      title: "Sensor States"
+      type: "[]string"
+      description: "Sensor states"
+      required: false
+    sensorVersion:
+      title: "Sensor Version"
+      type: string
+      description: "Sensor version"
+      required: false
+    status:
+      title: "Status"
+      type: string
+      description: "Status"
+      required: false
+    targetPriorityType:
+      title: "Target Priority Type"
+      type: string
+      description: "Target priority type"
+      required: false
+    testId:
+      title: "Test ID"
+      type: integer
+      description: "Test ID"
+      required: false
+    uninstallCode:
+      title: "Uninstall Code"
+      type: string
+      description: "Uninstall code"
+      required: false
+    uninstalledTime:
+      title: "Uninstalled Time"
+      type: string
+      description: "Uninstalled time"
+      required: false
+    vdiBaseDevice:
+      title: "VDI Base Device"
+      type: string
+      description: "VDI Base Device"
+      required: false
+    virtualMachine:
+      title: "Virtual Machine"
+      type: boolean
+      description: "Virtual machine"
+      required: false
+    virtualizationProvider:
+      title: "Virtualization Provider"
+      type: string
+      description: "Virtualization Provider"
+      required: false
+    windowsPlatform:
+      title: "Windows Platform"
+      type: string
+      description: "Windows platform"
+      required: false
+
+connection:
+  api_id:
+    title: API ID
+    type: string
+    description: API ID
+    example: ABCDE12345
+    required: true
+    order: 1
+  api_secret_key:
+    title: API Secret Key
+    type: credential_secret_key
+    description: API Secret Key
+    example: ABCDEFG12345ABCD1234ABCD
+    required: true
+    order: 2
+  url:
+    title: URL
+    type: string
+    description: API URL
+    default: https://api5.conferdeploy.net/integrationServices/v3/
+    example: https://api5.conferdeploy.net/integrationServices/v3/
+    required: true
+    order: 3
+actions:
+  get_agent_details:
+    title: Get Agent Details
+    description: Get Agent Details
+    input:
+      agent:
+        title: Agent
+        description: Agent to retrieve device information from. Accepts IP address, MAC address, hostname, or device ID. Search results are case sensitive
+        type: string
+        required: true
+        example: 198.51.100.1
+    output:
+      agent:
+        title: Agent
+        description: Details about the agent
+        type: agent
+        required: false
+      all_agents_matched:
```

Comment:
We should remove this input to only returning `agent`, it complicates the experience and would be hard to even use in a decision. The Search Agents action coming up will allow us to do search for multiple agents.

Reply:
This happens more often than it should. Their search API kind of sucks.

Reply:
How does it occur now? I think we only have 1 or 2 agents attached. We should only be returning the matching IP, MAC, hostname, or device ID. If you made the IP, hostname, or MAC the same on both, then we return the first matching one.

Plugin output goes from very simple, to complicated here (do I use agent or other_matches and how do I do that in a workflow to choose the right one) . Should save this logic for the upcoming Search Agents action which is designed for returning multiple endpoints.